### PR TITLE
cds-719 add workflow for release candidate

### PR DESCRIPTION
**Pull request by daidokoro**
- add workflow for release candidate
- add vscode to gitignore

# Description

 Fixes #

# How Has This Been Tested?

# Checklist:
- [ ] I have updated the versions in the SemanticVersion in template.yaml
- [ ] I have updated the CHANGELOG.md
- [ ] I have created necessary PR to Terraform Module Repository (https://github.com/coralogix/terraform-coralogix-aws) if needed
- [ ] This change does not affect any particular component (e.g. it's CI or docs change)

### CODE REVIEW

File: .github/workflows/publish-rc.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+      - name: package
+        run: |
+          sam package \
+            --s3-bucket ${{ env.AWS_SERVERLESS_BUCKET }} \
+            --s3-prefix ${{ env.AWS_S3_PREFIX }} \
+            --output-template-file packaged.yaml
+
+      - name: store
+        uses: actions/upload-artifact@v3
```
Comment (royfur) [inline .github/workflows/publish-rc.yaml:60]: need to update to v4

File: .github/workflows/publish-rc.yaml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+  publish:
+    name: publish
+    needs: build
+    if: ${{ github.event_name == 'workflow_dispatch' || success() }}
+    runs-on: ubuntu-latest
+    env:
+      AWS_SERVERLESS_BUCKET: coralogix-serverless-repo
+    steps:
+      - uses: aws-actions/configure-aws-credentials@v2
```
Comment (royfur) [inline .github/workflows/publish-rc.yaml:73]: need to update to v4

File: .github/workflows/publish-rc.yaml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID_STAGING }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY_STAGING }}
+          aws-region: ${{ env.AWS_DEFAULT_REGION }}
+
+      - name: download
+        uses: actions/download-artifact@v3
```
Comment (royfur) [inline .github/workflows/publish-rc.yaml:80]: need to update to v4

File: .github/workflows/publish-rc.yaml [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+          python-version: '3.9'
+          cache: 'pip' # caching pip dependencies
+      - run: pip3 install cargo-lambda
+
+      - uses: aws-actions/setup-sam@v2
+        with:
+          use-installer: true
+
+      - name: start ssh agent for rust private dependencies
+        uses: webfactory/ssh-agent@v0.8.0
```
Comment (royfur) [inline .github/workflows/publish-rc.yaml:40]: need to update to v0.9.0

**Review by juan-coralogix — APPROVED** [review]
LGTM. 
Not sure about the versions Roy mentioned in the other  Review.

**Pull request merged**